boa_ast/src/expression/operator/binary/mod.rs

```diff
@@ -38,7 +38,8 @@ pub struct Binary {
 
 impl Binary {
     /// Creates a `BinOp` AST Expression.
-    pub(in crate::syntax) fn new(op: BinaryOp, lhs: Expression, rhs: Expression) -> Self {
+    #[must_use]
```

Comment:
Why are we adding these attributes here? I understand this is meant to be used (like any result of any function), but it's not the same as a `Result` for example, which could contain important information of the execution.

If the user wants to warn against this, they have the `#[warn(unused)]` attribute.

Reply:
They're generally used when a method or function returns something that the user must handle to be at least useful. For example, `Vec::new` is marked as [`must_use`](https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#423). False positives generally occur for types with interior mutability. but the ast doesn't contain any types with that property, so it makes sense to mark them all with `must_use`.

Reply:
Also, if for some reason an user intentionally wants to call a `must_use` function, adding a `let _ =` before it is enough, which is arguably better for readability.

Reply:
Hmm yep, I guess this is fine, but then, this should be applied to all getter and constructor methods throughout the code-base. I guess we could enable the clippy lint for this. I'm OK to merge this as-is, but we should probably have an issue or a separate PR to do this throughout the code base.